scanpipe/apps.py

```diff
@@ -83,6 +100,22 @@ def register_pipeline(self, name, cls):
 
         self._pipelines[name] = cls
 
+    def register_pipeline_from_file(self, path):
+        """
+        Search for a Pipeline subclass in the provided file `path` and register it
+        when found.
+        """
+        module_name = path.stem
+        module = SourceFileLoader(module_name, str(path)).load_module()
+        pipeline_classes = inspect.getmembers(module, is_pipeline)
+
+        if len(pipeline_classes) > 1:
+            raise ImproperlyConfigured("Only 1 Pipeline class allowed per file.")
+
+        elif pipeline_classes:
+            pipeline_class = pipeline_classes[0][1]
+            self.register_pipeline(name=module_name, cls=pipeline_class)
```

Comment:
Are we checking that the name is new and unique and does not mask the existing pipelines?

Reply:
Yes: https://github.com/nexB/scancode.io/blob/7ade8de7c0f5fee295ba7472c4811a30e06f1ab9/scanpipe/apps.py#L96-L99